### PR TITLE
[CAFV-130] Upgrade API to v1beta2

### PR DESCRIPTION
**Pull request by abaruni**
## Description
upgrade api version to v1beta2 and complete any required automatic and manual conversions from v1alpha4 -> v1beta2 & v1beta1 -> v1beta2

```
➜  ~ kube get VCDCluster.v1alpha4.infrastructure.cluster.x-k8s.io
No resources found in default namespace.
➜  ~ kube get VCDCluster.v1beta1.infrastructure.cluster.x-k8s.io
No resources found in default namespace.
➜  ~ kube get VCDCluster.v1beta2.infrastructure.cluster.x-k8s.io
No resources found in default namespace.
➜  ~ kube get VCDMachine.v1alpha4.infrastructure.cluster.x-k8s.io
No resources found in default namespace.
➜  ~ kube get VCDMachine.v1beta1.infrastructure.cluster.x-k8s.io
No resources found in default namespace.
➜  ~ kube get VCDMachine.v1beta2.infrastructure.cluster.x-k8s.io
No resources found in default namespace.
➜  ~ kube get VCDMachineTemplate.v1alpha4.infrastructure.cluster.x-k8s.io
No resources found in default namespace.
➜  ~ kube get VCDMachineTemplate.v1beta1.infrastructure.cluster.x-k8s.io
No resources found in default namespace.
➜  ~ kube get VCDMachineTemplate.v1beta2.infrastructure.cluster.x-k8s.io
No resources found in default namespace.
```

## Checklist
- [x] tested locally
- [ ] updated any relevant dependencies
- [ ] updated any relevant documentation or examples

## API Changes
Are there API changes?
- [x] Yes
- [ ] No

If yes, please fill in the below

1. Updated conversions?
    - [x] Yes
    - [ ] No
    - [ ] N/A
2. Updated CRDs?
    - [x] Yes
    - [ ] No
    - [ ] N/A
3. Updated infrastructure-components.yaml?
    - [x] Yes
    - [ ] No
    - [ ] N/A
4. Updated `./examples/capi-quickstart.yaml`?
    - [ ] Yes
    - [x] No
    - [ ] N/A
5. Updated necessary files under `./infrastructure-vcd/v1.0.0/`?
   - [ ] Yes
   - [ ] No
   - [x] N/A

## Issue
If applicable, please reference the relevant issue

Fixes #

- - -
This change is [Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/374)

### CODE REVIEW

File: api/v1beta1/vcdcluster_conversion.go [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+import (
+	"github.com/vmware/cluster-api-provider-cloud-director/api/v1beta2"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
 
-func (*VCDClusterList) Hub() {}
+// ConvertTo converts this VCDCluster to the Hub version (v1beta2).
+func (src *VCDCluster) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1beta2.VCDCluster)
+	return Convert_v1beta1_VCDCluster_To_v1beta2_VCDCluster(src, dst, nil)
```
Comment (Anirudh9794) [inline api/v1beta1/vcdcluster_conversion.go:11]: This needs to have the logic to restore fields using `utilconversion.UnmarshalData(src, restored)`

File: api/v1beta1/vcdcluster_conversion.go [hunk 2]
```diff
@@ -1,5 +1,30 @@
...
+}
+
+// ConvertFrom converts from the Hub version (v1beta2) to this version (v1beta1).
+func (dst *VCDCluster) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*v1beta2.VCDCluster)
+	return Convert_v1beta2_VCDCluster_To_v1beta1_VCDCluster(src, dst, nil)
```
Comment (Anirudh9794) [inline api/v1beta1/vcdcluster_conversion.go:17]: All the `ConvertFrom()` functions should store the dst object in an annotation in src using `utilconversion.MarshalData(src, dst)`

**Review by sahithi — APPROVED** [review]
:lgtm: after addressing the comment and testing one successful cluster creation with VCDCluster objects in v1beta2 format

Please create another Jira task to test the below scenarios

Case1: CAPI yaml with core capi objects in v1beta1 and our objects in v1beta1
Case2: CAPI yaml with core capi objects in v1beta1 and our objects in v1beta2
Case3: CAPI yaml with core capi objects in v1alpha4 and our objects in v1alpha4

> Reviewed 40 of 43 files at r1, 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/374#-:-NN8265-5Nb3TlP6tHKJ:biapd1d)* status: all files reviewed, 3 unresolved discussions (waiting on @abaruni and @arunmk)

___
*[`config/crd/kustomization.yaml` line 2 at r2](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/374#-NN8-yVz6fieSt-PdXKU:-NN8-yVz6fieSt-PdXKV:b-d84dg) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/df9dbf84753671108a9fa1a7dc3afde0147ac2f3/config/crd/kustomization.yaml#L2)):*
> ```YAML
> commonLabels:
>   cluster.x-k8s.io/v1beta2: v1beta2
> ```

This is incorrect. We have not yet upgraded core capi to v1beta2. These labels dictate the API contract between core capi and capvcd. It says - any given version of core capi must use which version of capvcd.


It should be something like
  cluster.x-k8s.io/v1beta1: v1beta1_v1beta2

This needs to be tested out thoroughly. 
Case1: CAPI yaml with core capi objects in v1beta1 and our objects in v1beta1
Case2: CAPI yaml with core capi objects in v1beta1 and our objects in v1beta2
Please create another task for testing if you want to submit this PR.

**Review by abaruni — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/374)* status: 40 of 53 files reviewed, 3 unresolved discussions (waiting on @Anirudh9794, @arunmk, and @sahithi)

___
*[`api/v1beta1/vcdcluster_conversion.go` line 11 at r2](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/374#-NMnWQEfYfNhZ1lGGQeI-r2-11:-NOpvQrrEasBcHYWTGTI:b-896fix) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/df9dbf84753671108a9fa1a7dc3afde0147ac2f3/api/v1beta1/vcdcluster_conversion.go#L11)):*
<details><summary><i>Previously, Anirudh9794 wrote…</i></summary><blockquote>

This needs to have the logic to restore fields using `utilconversion.UnmarshalData(src, restored)`
</blockquote></details>

Done.
___
*[`api/v1beta1/vcdcluster_conversion.go` line 17 at r2](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/374#-NMnWQEfYfNhZ1lGGQeI-r2-17:-NOpvRYa2FOTSB6hRR_B:b-896fix) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/df9dbf84753671108a9fa1a7dc3afde0147ac2f3/api/v1beta1/vcdcluster_conversion.go#L17)):*
<details><summary><i>Previously, Anirudh9794 wrote…</i></summary><blockquote>

All the `ConvertFrom()` functions should store the dst object in an annotation in src using `utilconversion.MarshalData(src, dst)`
</blockquote></details>

Done.
___
*[`config/crd/kustomization.yaml` line 2 at r2](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/374#-NN8-yVz6fieSt-PdXKU:-NOpvWeO4iiVKlujSWJ_:b-896fix) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/df9dbf84753671108a9fa1a7dc3afde0147ac2f3/config/crd/kustomization.yaml#L2)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

This is incorrect. We have not yet upgraded core capi to v1beta2. These labels dictate the API contract between core capi and capvcd. It says - any given version of core capi must use which version of capvcd.


It should be something like
  cluster.x-k8s.io/v1beta1: v1beta1_v1beta2

This needs to be tested out thoroughly. 
Case1: CAPI yaml with core capi objects in v1beta1 and our objects in v1beta1
Case2: CAPI yaml with core capi objects in v1beta1 and our objects in v1beta2
Please create another task for testing if you want to submit this PR.
</blockquote></details>

Done.

**Review by Anirudh9794 — APPROVED** [review]
:lgtm: with a small comment

> Reviewed 32 of 43 files at r1, 7 of 8 files at r2, 13 of 13 files at r3, 20 of 20 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/374#-:-NOq9CvX41DGshn46JX7:b-q7lpd0)* status: all files reviewed, 4 unresolved discussions (waiting on @abaruni, @arunmk, and @sahithi)

___
*[`api/v1beta1/vcdmachinetemplate_conversion.go` line 27 at r3](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/374#-NOq6r_S8DYw-4YyI2WM:-NOq6r_S8DYw-4YyI2WN:bo38hqr) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/e6937664ba0bc01dcc563c1850a555138895e55c/api/v1beta1/vcdmachinetemplate_conversion.go#L27)):*
> ```Go
> func (dst *VCDMachineTemplate) ConvertFrom(srcRaw conversion.Hub) error {
> 	src := srcRaw.(*v1beta2.VCDMachineTemplate)
> 	return Convert_v1beta2_VCDMachineTemplate_To_v1beta1_VCDMachineTemplate(src, dst, nil)
> ```

`utilconversion.MarshalData(src, dst)` is missing from this function

**Pull request merged**